sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/predicates.scala

```diff
@@ -334,7 +318,7 @@ case class GreaterThan(left: Expression, right: Expression) extends BinaryCompar
 case class GreaterThanOrEqual(left: Expression, right: Expression) extends BinaryComparison {
   override def symbol: String = ">="
 
-  override protected def checkTypesInternal(t: DataType) =
+  protected def checkTypesInternal(t: DataType) =
     TypeUtils.checkForOrderingExpr(left.dataType, "operator " + symbol)
```

Comment:
i think we can remove checkForOrdering by defining either an ordering ADT, or define a constant using TypeCollection